build/README.md

```diff
@@ -160,6 +160,42 @@ class ResolvingCopyBuilder {
 Once you have gotten a `LibraryElement` using one of the methods on `Resolver`,
 you are now just using the regular `analyzer` package to explore your app.
 
+### Sharing expensive objects across build steps
+
+The build package includes a `Resource` class, which can give you an instance
+of an expensive object that is guaranteed to be unique across builds, but may
+be re-used by multiple build steps within a single build (to the extent that
+the implementation allows). It also gives you a way of disposing of your
+resource at the end of its lifecycle.
+
+The `Resource<T>` constructor takes a single required argument which is a
+factory function that returns a `FutureOr<T>`. There is also a named argument
+`dispose` which is called at the end of life for the resource, with the
+instance that should be disposed. This returns a `FutureOr<dynamic>`.
+
+So a simple example `Resource` would look like this:
+
+```dart
+final resource = new Resource(
+  () => createMyExpensiveResource(),
```

Comment:
[nit] unnecessary closure?

Reply:
for the docs though I think this is more clear what the argument is (unless I actually showed a fake impl of createExpensiveResource which I think is not necessary)

Reply:
fair enough